demos/saddle_point_pc/saddle_point_systems.py.rst

```diff
@@ -466,72 +460,61 @@ Instead of asking PETSc to build an approximation to :math:`S` which
 we then use to solve the problem, we can provide one ourselves.
 Recall that :math:`S` is spectrally a Laplacian only in a
 discontinuous space.  A natural choice is therefore to use an interior
-penalty DG formulation for the Laplacian term and provide it as
-:math:`Sp`.  Since this preconditioning matrix is block-diagonal, we
-then need to tell PETSc that it should use the original operator,
-rather than the preconditioning matrix, to compute the action of the
-off-diagonal blocks.  This is done using
-``pc_fieldsplit_off_diag_use_amat``. ::
-
+penalty DG formulation for the Laplacian term on the block of the scalar
+variable. We can provide it as an AuxiliaryOperator via a python preconditioner. ::
+
+    class DGLaplacian(AuxiliaryOperatorPC):
+        def form(self, pc, u, v):
+            W = u.function_space()
+            n = FacetNormal(W.mesh())
+            alpha = Constant(8.0)
+            gamma = Constant(8.0)
+            h = CellSize(W.mesh())
+            h_avg = (h('+') + h('-'))/2
+            a_dg =  -(inner(grad(v), grad(u))*dx 
```

Comment:
I think you don't want this minus sign?

Reply:
I think this form (if the old code is correct) should be
```
               + dot(grad(v), grad(u))*dx \
                - dot(avg(grad(v)), jump(u, n))*dS \
                - dot(jump(v, n), avg(grad(u)))*dS \
                + alpha/h_avg * dot(jump(v, n), jump(u, n))*dS \
                - dot(grad(v), u*n)*ds \
                - dot(v*n, grad(u))*ds \
                + (gamma/h)*dot(v, u)*ds
```

And then I suppose we have to update for complex-safety so

```
               + inner(grad(u), grad(v))*dx \
                - inner(jump(u, n), avg(grad(v)))*dS \
                - inner(avg(grad(u)), jump(v, n), )*dS \
                + alpha/h_avg * inner(jump(u, n), jump(v, n))*dS \
                - inner(u*n, grad(v))*ds \
                - inner(grad(u), v*n)*ds \
                + (gamma/h)*inner(u, v)*ds
```

Reply:
Okay, we do want the negative sign. The way the flux is introduced to bring the problem into mixed form, makes the the operator of u here negative definite. (Thanks @colinjcotter!)